[CI] OTelMetricsBufferingIT testOutageBuffersToDiskAndDrainsOnRecovery failing

**Build Scans:**
- [elasticsearch-intake #41243 / part6](https://gradle-enterprise.elastic.co/s/acsxujdjxjbuy)
- [elasticsearch-pull-request #153362 / part-6](https://gradle-enterprise.elastic.co/s/jkqetakmfk47a)

**Reproduction Line:**
```
./gradlew ":test:external-modules:test-apm-integration:javaRestTest" --tests "org.elasticsearch.test.apmintegration.OTelMetricsBufferingIT.testOutageBuffersToDiskAndDrainsOnRecovery" -Dtests.seed=8F0FF6810080F38 -Dtests.locale=zh-Hans -Dtests.timezone=Pacific/Marquesas -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.test.apmintegration.OTelMetricsBufferingIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testOutageBuffersToDiskAndDrainsOnRecovery'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: expected the drain loop to replay all disk-buffered batches after recovery (es.apm.metrics.disk_buffer.writes peaked at 0, es.apm.metrics.disk_buffer.replays peaked at 0)
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testOutageBuffersToDiskAndDrainsOnRecovery
- [main] 1 consecutive failures in step part6
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 2 failures in test testOutageBuffersToDiskAndDrainsOnRecovery (8.3% fail rate in 24 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -506,9 +506,6 @@ tests:
 - class: org.elasticsearch.xpack.ml.qa.MlYamlRestIT
   method: test {p0=ml/data_frame_analytics_crud/Test put valid config with default outlier detection, query, and filter}
   issue: https://github.com/elastic/elasticsearch/issues/150630
-- class: org.elasticsearch.test.apmintegration.OTelMetricsBufferingIT
-  method: testOutageBuffersToDiskAndDrainsOnRecovery
-  issue: https://github.com/elastic/elasticsearch/issues/150387
 - class: org.elasticsearch.xpack.esql.qa.ndjson.NdJsonCompressedFormatSpecIT
   method: test {csv-spec:external-basic.forkTwoBranches [ndjson.bz/AZURE]}
   issue: https://github.com/elastic/elasticsearch/issues/150651

--- a/test/external-modules/apm-integration/src/javaRestTest/java/org/elasticsearch/test/apmintegration/RecordingApmServer.java
+++ b/test/external-modules/apm-integration/src/javaRestTest/java/org/elasticsearch/test/apmintegration/RecordingApmServer.java
@@ -35,7 +35,8 @@ import java.net.InetAddress;
 import java.net.InetSocketAddress;
 import java.nio.charset.StandardCharsets;
 import java.util.List;
-import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.Consumer;
@@ -44,7 +45,7 @@ import java.util.function.Consumer;
 public class RecordingApmServer extends ExternalResource {
     private static final Logger logger = LogManager.getLogger(RecordingApmServer.class);
 
-    final ArrayBlockingQueue<ReceivedTelemetry> received = new ArrayBlockingQueue<>(1000);
+    private final BlockingQueue<ReceivedTelemetry> received = new LinkedBlockingQueue<>();
 
     /**
      * The "Resource" (telemetry source identity) observed by this server. The test JVM emits
@@ -157,6 +158,9 @@ public class RecordingApmServer extends ExternalResource {
                 }
             }
             exchange.sendResponseHeaders(responseCode, 0);
+        } catch (Throwable t) {
+            logger.error("Unexpected error caught when serving HTTP request", t);
+            throw t;
         }
     }
 

